src/Leaflet.Renderer.Canvas.Tile.js

```diff
@@ -29,7 +29,23 @@ L.Canvas.Tile = L.Canvas.extend({
 	},
 
 	getOffset: function() {
-		return this._tileCoord.scaleBy(this._size).subtract(this._map.getPixelOrigin());
+		return this._tileCoord
+			.scaleBy(this._size)
+			.multiplyBy(this.getOverZoomFactor())
+			.subtract(this._map.getPixelOrigin())
+			.divideBy(this.getOverZoomFactor());
+	},
+
+	getOverZoomFactor: function() {
+		var numberOfZoomLevels = Math.max(0, this._map.getZoom() - this._tileCoord.z);
```

Comment:
There is a case in leaflet fractional zooming where the tiles are "underzoomed". 
So `this._map.getZoom()` is smaller than `_tileCoord.z`.
By Removing the "0"-Limit "underzoomed" cases will work as well. 
```suggestion
		var numberOfZoomLevels = this._map.getZoom() - this._tileCoord.z;
```

This Case happens when using zoomSnap of 0.25. Fractions < 0.5 take overzoom, Fractions >= 0.5 underzoom